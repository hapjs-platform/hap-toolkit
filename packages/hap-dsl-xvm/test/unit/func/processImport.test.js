/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */


'use strict'
const path = require('@jayfate/path')
const { processImport } = require('@hap-toolkit/compiler/lib/style/process')
const testCssDir = path.resolve(__dirname, '../../case/ux/Helloworld/Common')

const code = `
/* comment */
@import './common.css' screen and (orientation: portrait);
.box {
  width: 100px;
}
@import './common2.css' screen and (orientation: portrait);

`

describe('测试processImport函数', () => {
  it('process import', async () => {
    const result = processImport(code, testCssDir, [], [])
    expect(result).toMatchSnapshot()
  })
})
