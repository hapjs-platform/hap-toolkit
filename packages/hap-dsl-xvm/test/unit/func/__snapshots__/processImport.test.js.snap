// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试processImport函数 process import 1`] = `
"/* comment */


@media  screen and (orientation: portrait) {
/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

/*@import  './common2.css';*/


/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */


@media  screen and (resolution: 96dpi) {
/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.commom3 {
  color: #000033;
}
}


.common1 {
  color: #000001;
}


.common {
  color: #000000;
}
}


.box {
  width: 100px;
}

@import './common2.css' screen and (orientation: portrait);"
`;
